juriscraper/opinions/united_states/federal_special/uscg_ccap.py

```diff
@@ -0,0 +1,46 @@
+# Scraper for US Coast Guard Court of Criminal Appeals
+
+from juriscraper.lib.string_utils import convert_date_string
+from juriscraper.OpinionSite import OpinionSite
+from requests.utils import requote_uri
+
+
+class Site(OpinionSite):
+    def __init__(self, *args, **kwargs):
+        super(Site, self).__init__(*args, **kwargs)
+        self.court_id = self.__module__
+        self.current_page = 1
+        self.base_url = (
+            "https://www.uscg.mil/Resources/Legal/"
+            "Court-of-Criminal-Appeals/"
+            "CGCCA-Opinions/smdpage15701/{}"
+        )
+
+    def __format_url(self):
+        self.url = self.base_url.format(self.current_page)
+
+    def _process_html(self):
+        table_xpath = "//table[@class='Dashboard']//tbody[1]"
+        self.html = self.html.xpath(table_xpath)[0]
+
+    def _get_case_dates(self):
+        dates_xpath = "//tr//td[3]"
+        return [
+            convert_date_string(date.text_content())
+            for date in self.html.xpath(dates_xpath)
+        ]
+
+    def _get_case_names(self):
+        names_xpath = "//tr//td[2]"
+        return [name.text_content() for name in self.html.xpath(names_xpath)]
+
+    def _get_precedential_statuses(self):
+        rows_xpath = "//tr//td[1]"
+        return ["Published"] * (len(self.html.xpath(rows_xpath)))
+
+    def _get_download_urls(self):
+        # All of the case urls contain spaces,
+        # encoding them to be a good citizen.
+        url_xpath = "//tr//td[1]/a/@href"
+
+        return [requote_uri(url) for url in self.html.xpath(url_xpath)]
```

Comment:
cool `requote_uri`!  I wasn't aware of this.

Reply:
Neither was I haha.

Reply:
were the URLs parsed from the page wonky without running them through this function?  If so, @mlissner, I'm wondering if it might be helpful to run this function against all urls up in `AbstractSite`

Reply:
The URLs contain spaces. Isn't necessarily an issue and the links resolve in Firefox with and without encoding. It's just a nice to have in case someone needs to work with them programmatically or if some browser doesn't handle it.